driver-core/src/main/com/mongodb/operation/CommandOperationHelper.java

```diff
@@ -222,6 +222,10 @@ public T call(final ConnectionSource source) {
         } catch (MongoException e) {
             exception = e;
 
+            if (e instanceof MongoSocketException) {
```

Comment:
I think the check is missing for sync and async write commands.

I suggest adding a few more tests to the YAML file, and then submitting it upstream to specifications

Specifically, add separate tests for:

* a non-bulk write command, e.g. findOneAndUpdate
* a non-cursor returning read command, e.g. countDocuments

That should cover the places were the java driver is missing checks.

Reply:
Added tests are in code review ([SPEC-1503](https://jira.mongodb.org/browse/SPEC-1503)).